indico/modules/categories/forms.py

```diff
@@ -119,57 +142,103 @@ def __init__(self, *args, category, **kwargs):
                 )
-                return False
-            elif res == GoogleCredentialValidationResult.refused:
-                self.google_wallet_credentials.errors.append(
-                    _('The credentials are invalid or have no access to the Google Wallet API.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.failed:
-                self.google_wallet_credentials.errors.append(
-                    _('There was an error validating your credentials. Contact your Indico administrator for details.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.bad_issuer:
-                self.google_wallet_issuer_id.errors.append(
-                    _('The Issuer ID is invalid or not linked to your credentials.')
-                )
-                return False
+            ):
+                res = GoogleWalletManager.verify_credentials(credentials, issuer_id)
+                if res == GoogleCredentialValidationResult.invalid:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials could not be loaded.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.refused:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials are invalid or have no access to the Google Wallet API.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.failed:
+                    self.google_wallet_credentials.errors.append(
+                        _('There was an error validating your credentials. Contact your Indico administrator for '
+                          'details.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.bad_issuer:
+                    self.google_wallet_issuer_id.errors.append(
+                        _('The Issuer ID is invalid or not linked to your credentials.')
+                    )
+                    return False
 
         return form_valid
 
+    def validate_apple_pass_certificate(self, field):
+        try:
+            x509.load_pem_x509_certificate(field.data.encode('UTF-8'))
+            return True
+        except ValueError:
+            raise ValidationError(_('The provided Certificate.pem is malformed.'))
+
+    def validate_apple_pass_key(self, field):
+        try:
+            serialization.load_pem_private_key(field.data.encode('UTF-8'), password=None)
+        except ValueError:
+            raise ValidationError(_('The provided Private.key is malformed.'))
+        except TypeError:
+            # This is because I provided an empty pass, just to check the Private.key is not malformed.
+            # Is there a better way to check it?
+            return True
```

Comment:
The `TypeError` here is raised if the pass used to decrypt the private key is not correct, and this is obvious because I'm using and empty password.
But if so, this means that the private key is correctly recognized.

Reply:
I would rephrase the comment and move this `except` block to the at the bottom of this method.

```suggestion
            # TypeError is raised when no password is provided for the encrypted key, 
            # which we don't need to decrypt to check that it's malformed.
            return True
```

Reply:
Done.